uint16 u1
uint16 u2
uint16 D1
uint16 D2
